osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs

```diff
@@ -1,31 +1,63 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using System.Linq;
+using osu.Framework.Bindables;
 using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
 using osu.Game.Rulesets.Mods;
 using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Scoring;
 using osu.Game.Rulesets.UI;
+using osu.Game.Screens.Play;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModAlternate : ModAlternate<OsuHitObject, OsuAction>
+    public class OsuModAlternate : Mod, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToPlayer, IUpdatableByPlayfield
     {
+        public override string Name => @"Alternate";
+        public override string Acronym => @"AL";
+        public override string Description => @"Don't use the same key twice in a row!";
+        public override double ScoreMultiplier => 1.0;
+        public override Type[] IncompatibleMods => new[] { typeof(ModAutoplay) };
+        public override ModType Type => ModType.DifficultyIncrease;
+        public override IconUsage? Icon => FontAwesome.Solid.Keyboard;
+
+        /// <summary>
+        /// Whether incoming input must be checked by <see cref="InputInterceptor"/> before it is passed to gameplay.
+        /// </summary>
+        public bool CanIntercept => !isBreakTime.Value && introEnded;
```

Comment:
i'm not sure this is much better/clearer than before. can we just rename this to something obvious like `PlayerMustAlternate` or something?

also, i forgot about it before, but remembered now but some maps have storyboarded outros (and lazer supports this already) - so outro start should probably get checked here, too.

Reply:
There's no hit objects in storyboard outros so this should be sufficient.
I also do agree the xmldoc still doesn't help. I'm bad at naming and explaining things so I'll take up on your suggestion.

Reply:
sure, but the cursor will still flash red if you happen to spam keys in outro, would it not? unless you're saying that storyboard outro is also a break (which it may well be, i can't say i went and checked).

Reply:
never mind, i checked and it works as expected, because `IsBreakTime` also includes score processor completion for some reason

https://github.com/ppy/osu/blob/4132f67629467f5a5f2b2b480b391525bfe7d95e/osu.Game/Screens/Play/BreakTracker.cs#L56-L58